### DISCUSSION TITLE
How can I add nitro routeRules for dynamic pages?

### DISCUSSION THREAD

**Discussion by lionel-addvanto** [nuxt/nuxt#29887] [Questions]
If i have `pages/[slug].vue` as a file, how can I add a `routeRule` for it in the `nuxt.config.ts`?

**Comment by logotip4ik** [comment]
I would use `defineRouteRules` https://nuxt.com/docs/api/utils/define-route-rules

**Reply by oneminch** [reply]
I think the best approach would be to use `defineRouteRules` in your `[slug].vue` as @logotip4ik suggested, but you need to allow this in your config using the `experimental.inlineRouteRules` option since it is experimental.

```vue
<script setup lang="ts">
  defineRouteRules({
    ssr: true,
    prerender: false,
    cache: { maxAge: 60 * 60 }
  });
</script>
```

```ts
export default defineNuxtConfig({
  experimental: {
    inlineRouteRules: true
  }
})
```

You can also define route rules in your config by matching the page pattern. But, it's likely that the `/**` pattern would match all nested routes.

```ts
export default defineNuxtConfig({
  // ...
  routeRules: {
    // pages/[slug].vue
    '/**': { prerender: true },
    // pages/blog/[slug].vue
    '/blog/**': { prerender: true }
  }
});
```
